odp_sdk/tabular_storage_client.py

```diff
@@ -254,15 +267,128 @@ def delete_stage_request(self, resource_dto: ResourceDto, table_stage: TableStag
         except requests.HTTPError as e:
             if response.status_code == 400:  # This is coming as 400 for both 400 and 404 from ODP
                 raise OdpResourceNotFoundError("Schema not found") from e
+            raise
+
+    def select(self, resource_dto: ResourceDto, filter_query: Optional[dict]) -> List[Dict]:
+        """
+        Select data from dataset
+
+        Args:
+            resource_dto: Dataset manifest
+            filter_query: filter query in OQS format
+
+        Returns:
+            Data that is queried
+
+        Raises
+            OdpResourceNotFoundError: If the schema cannot be found
+        """
+
+        if resource_dto.metadata.uuid:
+            url = f"{self.tabular_storage_url}/{resource_dto.metadata.uuid}/list"
+        else:
+            url = f"{self.tabular_storage_url}/catalog.hubocean.io/dataset/{resource_dto.metadata.name}/list"
+
+        if filter_query:
+            response = self._http_client.post(url, content=filter_query)
+        else:
+            response = self._http_client.post(url)
+
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as e:
+            if response.status_code == 404:
+                raise OdpResourceNotFoundError("Resource not found") from e
+            raise
 
-    def select(self, resource_dto: ResourceDto, filter_query: Optional[OqsBasePredicate]) -> List[Dict]:
-        raise NotImplementedError()
+        return [row for row in response.json()]
 
     def write(self, resource_dto: ResourceDto, data: List[Dict], table_stage: Optional[TableStage]):
-        raise NotImplementedError()
+        """
```

Comment:
What happens if the `data`-list is very long (say `> 1_000_000`) rows? Maybe this function can do chunked writes?